### PR TITLE
Concretize gadgets for optional custom gates

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR removes the first-class module arguments to functions from the gadgets for optional custom gates. This PR allows `Snarky_impl.Run.Typ` types to be opaque; without this we see type escape errors.

This code is unused, so this is effectively a no-op.

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by volhovm — APPROVED** [review]

File: src/lib/crypto/kimchi_backend/gadgets/bitwise.mli [hunk 1]
```diff
@@ -1,3 +1,5 @@
+module Circuit := Kimchi_pasta_snarky_backend.Step_impl
```
Comment (querolita) [inline src/lib/crypto/kimchi_backend/gadgets/bitwise.mli:1]: It reads much cleaner now

**Review by querolita — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**